src/config.ml

```diff
@@ -146,8 +146,19 @@ module Conf_map = struct
          | None | Some `Server -> Error "is not a TLS client"
          | Some `Client -> Ok ()) >>= fun () ->
         let _todo = ensure_not in
-        ensure_mem Auth_user_pass "does not have user/password"
-        (* ^-- TODO or has client certificate ? see options.c:--cert/--key,--pkcs12,or--auth-user-pass *)
+        begin match find Auth_user_pass t,
+                    find Tls_cert t,
+                    find Tls_key t
+          with
```

Comment:
my intuition would be that there are two cases that are fine: `Auth_user_pass` is `Some _` OR _both_ `Tls_cert` and `Tls_key` are provided. I would not know what to do if all three are provided (does it make sense to authenticate with a TLS cert + key and user/password? - does anyone do that?)